app/src/main/java/org/simple/clinic/scheduleappointment/ScheduleAppointmentModel.kt

```diff
@@ -2,11 +2,32 @@ package org.simple.clinic.scheduleappointment
 
 import android.os.Parcelable
 import kotlinx.android.parcel.Parcelize
+import org.simple.clinic.overdue.PotentialAppointmentDate
+import org.simple.clinic.overdue.TimeToAppointment
+import org.simple.clinic.util.UserClock
 
 @Parcelize
-class ScheduleAppointmentModel : Parcelable {
+class ScheduleAppointmentModel(
+    val potentialAppointmentDates: List<PotentialAppointmentDate>
+) : Parcelable {
 
   companion object {
-    fun create(): ScheduleAppointmentModel = ScheduleAppointmentModel()
+    fun create(
+        timeToAppointments: List<TimeToAppointment>,
+        userClock: UserClock
+    ): ScheduleAppointmentModel {
+      return ScheduleAppointmentModel(
+          potentialAppointmentDates = generatePotentialAppointmentDatesForScheduling(timeToAppointments, userClock)
+      )
+    }
+
+    private fun generatePotentialAppointmentDatesForScheduling(
+        timeToAppointments: List<TimeToAppointment>,
+        clock: UserClock
+    ): List<PotentialAppointmentDate> {
+      return PotentialAppointmentDate.from(timeToAppointments, clock)
+          .distinctBy(PotentialAppointmentDate::scheduledFor)
+          .sorted()
```

Comment:
Why can't we move this generation to the sheet itself and just initialize the model with the `potentialAppointmentDates`? 
We are anyway injecting `userClock` in the sheet. Passing that to the `Model` and then letting it generate this data — does this have any advantage?

Reply:
We could move this, I shall do this.

Reply:
Or wait, I have decided not to do this because I would have duplicate this code in `ScheduleAppointmentSheetControllerTest` as well.